### PR TITLE
Feature/issue 45

### PR DESCRIPTION
**Pull request by jzaeske**
#45

### CODE REVIEW

**Comment by mikaelcom** [comment]
Would it be possible to you to add a unit test with the generation of a Struct class in StructTest?
This test should match your improvement. If you want I can do it if you provide me the required WSDL/XSD.

Let me know, thx

File: src/Model/AbstractModel.php [hunk 1]
```diff
@@ -275,10 +275,14 @@ public function getPackagedName($namespaced = false)
         if ($namespaced && $this->getNamespace() !== '') {
             $nameParts[] = sprintf('\%s\\', $this->getNamespace());
         }
+        $cleanName = $this->getCleanName();
         if ($this->getGenerator()->getOptionPrefix() !== '') {
             $nameParts[] = $this->getGenerator()->getOptionPrefix();
+        } else {
+            $cleanName = self::replaceReservedPhpKeyword($cleanName, null);
```
Comment (mikaelcom) [inline src/Model/AbstractModel.php]: As you defined `$context` as `null` by default at line [366](https://github.com/WsdlToPhp/PackageGenerator/pull/46/files#diff-96f379c0a0aae9c0c450923d7d3b98b3R366), why do you pass `null`as second parameter here ?

**Comment by jzaeske** [comment]
> If you want I can do it if you provide me the required WSDL/XSD.
Unfortunately, I'm not allowed to provide our customers WSDL. I will try to create a miminal example which matches the problem case.

**Comment by mikaelcom** [comment]
You can use the PayPal tests resource as it has a generated Do Service class.
Another example is the Acton service contains a an operation named list which should generate a List Service class.

File: src/Model/AbstractModel.php (hunk 1 above)
Comment (jzaeske) [inline src/Model/AbstractModel.php]: If null Arguments trigger a special handling I like them to be called explicitly. I think that makes the code more readable.

**Comment by jzaeske** [comment]
Thanks for this Information. I have adapted the Paypal Generation Test without the 'Api'-Prefix to test Generation for Class '_Do'

**Comment by mikaelcom** [comment]
:+1: thx,

I'll review it very soon

**Comment by mikaelcom** [comment]
my bad!

**Pull request merged**